packages/mantine/src/components/table/Table.tsx

```diff
@@ -123,6 +137,23 @@ export const Table = <T,>(props: TableProps<T> & {ref?: ForwardedRef<HTMLDivElem
     const footer = convertedChildren.find((child) => child.type === TableFooter);
     const lastUpdated = convertedChildren.find((child) => child.type === TableLastUpdated);
     const noData = convertedChildren.find((child) => child.type === TableNoData);
+    const actions = convertedChildren.find((child) => child.type === TableActions);
+
+    const actionsForHeader = !!actions ? cloneElement(actions, {...actions.props, variant: 'split'}) : null;
+    // clone the header and add the actions to it
+    const headerWithActions = !!header
+        ? cloneElement(header, {
+              ...header.props,
+              children: !!actions ? (
+                  <>
+                      {header.props.children}
+                      <TableHeaderActions>{actionsForHeader}</TableHeaderActions>
+                  </>
+              ) : (
+                  header.props.children
+              ),
+          })
+        : null;
```

Comment:
We could enhance the logic here so that we automatically add the `<Table.Header />` if it is missing. I am thinking of something like this (naming to review perhaps).

```suggestion
    let headerWithActions = null;
    if (!!header && !!actionsForHeader) {
        headerWithActions = cloneElement(header, {
            ...header.props,
            children: (
                <>
                    {header.props.children}
                    <TableHeaderActions>{actionsForHeader}</TableHeaderActions>
                </>
            ),
        });
    } else if (!header && !!actionsForHeader) {
        headerWithActions = (
            <Table.Header>
                <TableHeaderActions>{actionsForHeader}</TableHeaderActions>
            </Table.Header>
        );
    } else if (!!header && !actionsForHeader) {
        headerWithActions = header;
    }
```

Reply:
I simplified it a bit since I can directly use `actions` instead of `actionsForHeader`. Do you think it's still worth it to automatically add the header?

Reply:
Does this mean we can avoid doing this now? 
<img width="224" alt="image" src="https://github.com/coveo/plasma/assets/35579930/7317d066-1ef3-49b1-8cf7-3821b5634cf5">

Reply:
We already have some tables with the actions in the rows without a `Table.Header` in the admin-ui. I think it's a legit case

Reply:
Ah yeah ok makes sense